// dao/modules/daoChr.js

	exports.create = function(mongo) {

		var	Schema		= mongo.Schema,
			ObjectId	= Schema.ObjectId;

		var chrSchema	= new Schema({
				id		: ObjectId,
				name	: String,
				data	: {}
			});

		var Chr = mongo.model('Chr',chrSchema);
			
		return {
			insert: function(req,res) {

				var data = req.body.data;	

				var chr = new Chr();

				chr.name	= data.name;
				chr.data 	= data;

				chr.save(function() {
					
					console.log("chr save");
					
					Chr.find({},function(err, coll) {
						if(coll) {
							var data = [];

							coll.forEach(function(el, ind) {
								data.push({id: coll[ind]._id, name: coll[ind].name});
							});

							res.send({coll: data});
						} else {	
							res.send({coll: null});
						}
						
					});
				});

			},

			select: function(req,res) {
				
            	Chr.find({},function(err, coll) {
			
					if(coll) {
						console.log("chr list");
						var data = [];

						coll.forEach(function(el, ind) {
							data.push({id: coll[ind]._id, name: coll[ind].name});
						});

						res.send({coll: data});
					} else {	
						res.send({coll: null});
					}
				});
				
			},

			selectData: function(id,res) {
				var id = id;
				
				Chr.findOne({"_id" : id}, function(err, coll) {
					
					if(coll) {
						console.log("chr select");
												
						res.send({id: coll._id, coll: coll});

					} else {
						res.send({coll: null});
					}
					
				});

			},
			
			update: function(req,res) {
				var id		= req.body.id,
					data	= req.body.data;

				Chr.findOne({"_id" : id}, function(err, coll) {
						coll.name = data.name;
						coll.data = data;
						
						coll.save(function() {
							console.log("update");
							
							Chr.find({},function(err, coll) {
								if(coll) {
									var data = [];
			
									coll.forEach(function(el, ind) {
										data.push({id: coll[ind]._id, name: coll[ind].name});
									});
			
									res.send({coll: data});
								} else {	
									res.send({coll: null});
								}
							});
							
						});
				});
			},

			remove: function(req,res) {
				var id = req.body.id;
				
				Chr.remove({"_id" : id}, function(){
					console.log("remove");
					
					Chr.find({},function(err, coll) {
						if(coll) {
							var data = [];

							coll.forEach(function(el, ind) {
								data.push({id: coll[ind]._id, name: coll[ind].name});
							});

							res.send({coll: data});
						} else {	
							res.send({coll: null});
						}
					});
				});
			},
			
			getModel: function() {
				return Chr;
			}
		}
	}
